### ISSUE TITLE
EC11 Encoder

### ISSUE THREAD

**Issue by Gabbajoe** [k1ngjet3r/PicoPad#1]
Hi have you debounce the EC11 rotary encoder like this?
https://electronics.stackexchange.com/questions/512042/does-ec11-incremental-encoder-need-hardware-debouncing-and-vcc-for-encoder

**Comment by jeterlin-stripe** [comment]
Hi Gabbajoe, 

I have not considered about this before. The encoder on the keypad I built has been working without any problems for a while.

Thanks for sharing this to me and please share any your concerns regarding to this denouncing issue.

Best,
Jeter

**Comment by Gabbajoe** [comment]
Hi I tested it now with debounce and it made the scrolling sluggish and delayed so it seems that th kmk lib has some software compensation already build in. Just wiring the ec11 encoder directly to the pico had the best result.

**Comment by jeterlin-stripe** [comment]
Hi again, I think this form can provide you some answers to your questions: 

https://forum.arduino.cc/t/debounce-rotary-encoder/967691

Seems like the Kmk has baked something like this into the firmware. I tried to find the related info on the Kmk official document, but so far I did not see anything on it. I hope this could give you some closure about why the debouching doesn't work in this case.
